plugins/inputs/intel_powerstat/services/msr.go

```diff
@@ -0,0 +1,297 @@
+// +build linux
+
+package services
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs/intel_powerstat/data"
+	"golang.org/x/sync/errgroup"
+)
+
+const (
+	systemCPUPath                      = "/sys/devices/system/cpu/"
+	cpuCurrentFreqPartialPath          = "/sys/devices/system/cpu/cpu%s/cpufreq/scaling_cur_freq"
+	msrPartialPath                     = "/dev/cpu/%s/msr"
+	c3StateResidencyLocation           = "0x3FC"
+	c6StateResidencyLocation           = "0x3FD"
+	c7StateResidencyLocation           = "0x3FE"
+	maximumFrequencyClockCountLocation = "0xE7"
+	actualFrequencyClockCountLocation  = "0xE8"
+	throttleTemperatureLocation        = "0x1A2"
+	temperatureLocation                = "0x19C"
+	// TimestampCounterLocation is an MSR offset.
+	TimestampCounterLocation = "0x10"
+)
+
+// MsrService is responsible for interactions with MSR.
+type MsrService interface {
+	OpenAndReadMsr(core string) error
+	RetrieveCPUFrequencyForCore(core string) (float64, int64, error)
+	GetCpuIDs() []string
+	GetThrottleTemperature(CPU string) uint64
+	GetTemperature(CPU string) uint64
+	GetTimestampDelta(CPU string) uint64
+	GetMperfDelta(CPU string) uint64
+	GetAperfDelta(CPU string) uint64
+	GetC6Delta(CPU string) uint64
+	GetC3Delta(CPU string) uint64
+	GetC7Delta(CPU string) uint64
+	GetReadDate(CPU string) int64
+	SetReadDate(CPU string, date int64)
```

Comment:
Get rid of the getter/setter and use the exported fields directly.

Reply:
Done.